From b368a7a1b55cdf53268e1f3cff3dc245bf5813c8 Mon Sep 17 00:00:00 2001
From: Richard Diamond <wichard@vitalitystudios.com>
Date: Sat, 12 Mar 2016 17:45:49 -0600
Subject: [PATCH] Add Pepper API specific handles for windowing.

---
 include/vlc_vout_window.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/vlc_vout_window.h b/include/vlc_vout_window.h
index 63351ba..c36c463 100644
--- a/include/vlc_vout_window.h
+++ b/include/vlc_vout_window.h
@@ -53,6 +53,7 @@ enum {
     VOUT_WINDOW_TYPE_NSOBJECT,
     VOUT_WINDOW_TYPE_ANDROID_NATIVE,
     VOUT_WINDOW_TYPE_WAYLAND,
+    VOUT_WINDOW_TYPE_PPAPI_G3D,
 };
 
 /**
@@ -109,12 +110,14 @@ struct vout_window_t {
         void     *nsobject;      /* Mac OSX view object */
         void     *anativewindow; /* Android native window. */
         struct wl_surface *wl;   /* Wayland surface */
+        int32_t  pp_context;     /* PPAPI Graphics3D context */
     } handle;
 
     /* display server (mandatory) */
     union {
         char     *x11; /* X11 display (NULL = use default) */
         struct wl_display *wl;   /* Wayland struct wl_display pointer */
+        int32_t pp_instance; /* PPAPI Instance */
     } display;
 
     /* Control on the module (mandatory)
-- 
2.5.0

