From 1ff63e41f37c0c02bd8e09c7c031e97e45c8ada2 Mon Sep 17 00:00:00 2001
From: Richard Diamond <wichard@vitalitystudios.com>
Date: Thu, 24 Mar 2016 14:11:22 -0500
Subject: [PATCH 3/3] Make VLC work when checked out as a submodule.

---
 src/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index 5ee185f..9abc913 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -579,7 +579,7 @@ $(srcdir)/revision.txt:
 stamp-revision:
 	$(AM_V_at)rm -f -- revision.tmp
 	$(AM_V_GEN)if ! git \
-			--git-dir="$(top_srcdir)/.git/" describe \
+			--git-dir="`git rev-parse --git-dir`" describe \
 			--tags --long --match '?.*.*' --always; then \
 		cat $(srcdir)/revision.txt ; \
 	fi > revision.tmp
-- 
2.5.0

