From 058b20ae28d079ceda10c5b77603b290dd6a83af Mon Sep 17 00:00:00 2001
From: Richard Diamond <wichard@vitalitystudios.com>
Date: Thu, 24 Mar 2016 14:33:35 -0500
Subject: [PATCH] NaCl needs sys/uio.h for struct iovec.

---
 src/network/tls.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/network/tls.c b/src/network/tls.c
index 5787051..93a4b05 100644
--- a/src/network/tls.c
+++ b/src/network/tls.c
@@ -35,6 +35,7 @@
 #include <errno.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <sys/uio.h>
 
 #include <vlc_common.h>
 #include "libvlc.h"
-- 
2.5.0

