From 9b74eaaa72096a7bf69b451e95f7da457effb893 Mon Sep 17 00:00:00 2001
From: Richard Diamond <wichard@vitalitystudios.com>
Date: Thu, 24 Mar 2016 21:08:33 -0500
Subject: [PATCH] Add missing files to libvlccore.

---
 src/Makefile.am | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/Makefile.am b/src/Makefile.am
index a687d06..d202d8b 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -319,6 +319,7 @@ SOURCES_libvlc_os2 = \
 SOURCES_libvlc_nacl = \
 	posix/dirs.c \
 	nacl/cancel.c \
+	nacl/cond.c \
 	nacl/mutex.c \
 	nacl/plugin.c \
 	nacl/rwlock.c \
@@ -332,6 +333,7 @@ SOURCES_libvlc_nacl = \
 	posix/timer.c \
 	posix/linux_cpu.c \
 	posix/linux_specific.c \
+	posix/specific.c \
 	posix/rand.c \
 	$(NULL)
 
-- 
2.5.0

