From 7d2617693dc90cc648303b02fa8e336415937eb2 Mon Sep 17 00:00:00 2001
From: Richard Diamond <wichard@vitalitystudios.com>
Date: Sun, 12 Jun 2016 13:48:42 -0500
Subject: [PATCH] Only include the version control revision string in
 libvlccore.

---
 lib/core.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/lib/core.c b/lib/core.c
index fa5c4de..849887a 100644
--- a/lib/core.c
+++ b/lib/core.c
@@ -36,8 +36,6 @@
 #include <limits.h>
 #include <assert.h>
 
-#include "../src/revision.c"
-
 libvlc_instance_t * libvlc_new( int argc, const char *const *argv )
 {
     libvlc_threads_init ();
-- 
2.7.4

